### PR TITLE
Add rerank feature and some small fixes

### PR DESCRIPTION
**Pull request by yasufumi0410**
This relates #166

### CODE REVIEW

**Comment by yasufumi0410** [comment]
I have found that I will need to update a lot.
So, I discard this PR now.

**Comment by yasufumi0410** [comment]
I have fixed some critical issues about getting Ranker in reranking.

File: ivy-jars.properties [hunk 1]
```diff
@@ -30,14 +30,18 @@ ivy.jar.jackson-databind.name=jackson-databind
 ivy.jar.jackson-annotations.org=com.fasterxml.jackson.core
 ivy.jar.jackson-annotations.name=jackson-annotations
 
+ivy.jar.commons.org=org.apache.commons
+ivy.jar.commons-lang.rev=3.7
+ivy.jar.commons-lang.name=commons-lang3
+
+ivy.jar.commons-logging.rev=1.2
+ivy.jar.commons-logging.org=commons-logging
+ivy.jar.commons-logging.name=commons-logging
```
Comment (kojisekig) [inline ivy-jars.properties]: Do you need commons logging? I cannot see you use it in the patch...

**Comment by yasufumi0410** [comment]
Thank you for pointing out. I checked ivy.xml and delete unnecessary dependencies.

**Comment by kojisekig** [comment]
Okay but how about commons lang? Are you sure you're using it?

**Comment by yasufumi0410** [comment]
I have checked all dependencies to pass `ant clean package test` in root dir and `ant clean package` in ltr4l-solr dir.

**Comment by yasufumi0410** [comment]
I found that commons-logging is required in HttpClients. Without that, I faced following error when running FeatureExtract class.
`Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/logging/LogFactory
	at org.apache.http.conn.ssl.DefaultHostnameVerifier.<init>(DefaultHostnameVerifier.java:82)
	at org.apache.http.impl.client.HttpClientBuilder.build(HttpClientBuilder.java:955)
	at org.apache.http.impl.client.HttpClients.createDefault(HttpClients.java:56)
	at org.ltr4l.lucene.solr.client.FeatureExtractor.postTrainingData(FeatureExtractor.java:65)
	at org.ltr4l.lucene.solr.client.FeatureExtractor.execute(FeatureExtractor.java:56)
	at org.ltr4l.cli.FeatureExtract.extract(FeatureExtract.java:114)
	at org.ltr4l.cli.FeatureExtract.main(FeatureExtract.java:59)`

**Pull request merged**